packages/loader/container-loader/src/containerContext.ts

```diff
@@ -157,18 +157,22 @@ export class ContainerContext implements IContainerContext {
         return this._disposed;
     }
 
-    private readonly fluidModuleP = new LazyPromise<IFluidModule>(async () => {
-        const fluidModule = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "CodeLoad" },
+    private readonly fluidModuleP = new LazyPromise<IFluidModuleWithDetails>(async () => {
+        const loadCodeResult = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "CodeLoad" },
             async () => this.codeLoader.load(this.codeDetails),
         );
 
-        return fluidModule;
+        const hasDetails = (result: IFluidModuleWithDetails | IFluidModule): result is IFluidModuleWithDetails => {
```

Comment:
i wouldn't inline this function

Reply:
The idea was to avoid polluting the module with a function that only is used here.

Reply:
you can still have the function in the file and not export it

Reply:
Refactored.